annif/backend/backend.py

```diff
@@ -17,7 +25,12 @@ class AnnifBackend(metaclass=abc.ABCMeta):
 
     DEFAULT_PARAMETERS = {"limit": 100}
 
-    def __init__(self, backend_id, config_params, project):
+    def __init__(
+        self,
+        backend_id: str,
+        config_params: dict[str, Any] | SectionProxy,
```

Comment:
This union could be replaced by a [MutableMapping](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableMapping), but the current `dict[str, Any] | SectionProxy` union seemed more clear.

Reply:
Maybe we should convert objects from configparser module into plain dicts before passing them into the backend? I think that would be cleaner